lib/smart_proxy_remote_execution_ssh/session.rb

```diff
@@ -30,7 +30,14 @@ def initialize_command
       end
       output_path = File.join(File.dirname(remote_script), 'output')
 
-      @connector.async_run("#{su_prefix}#{remote_script} | /usr/bin/tee #{output_path}") do |data|
+      # pipe the output to tee while capturing the exit code
+      script = <<-SCRIPT
+      exec 4>&1
```

Comment:
when I use heredoc, I either remove indentation or indent it, this way I think it keeps 6 spaces in begining of each line so either 0 or 8?